modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/http4s/WebServerLauncher.scala

```diff
@@ -81,31 +83,44 @@ object WebServerLauncher extends ServerApp with LogInitialization {
   /**
     * Configures and builds the web server
     */
-  def webServer(as: AuthenticationService, q: engine.EventQueue, se: SeqexecEngine): Kleisli[Task, WebServerConfiguration, Server] = Kleisli { conf =>
+  def webServer(as: AuthenticationService, events: (engine.EventQueue, Topic[SeqexecEvent]), se: SeqexecEngine): Kleisli[Task, WebServerConfiguration, Server] = Kleisli { conf =>
     val logger = Logger.getLogger(getClass.getName)
     logger.info(s"Start server on ${conf.devMode ? "dev" | "production"} mode")
 
     BlazeBuilder.bindHttp(conf.port, conf.host)
       .withWebSockets(true)
       .mountService(new StaticRoutes(conf.devMode).service, "/")
-      .mountService(new SeqexecCommandRoutes(as, q, se).service, "/api/seqexec/commands")
-      .mountService(new SeqexecUIApiRoutes(as, q, se).service, "/api")
+      .mountService(new SeqexecCommandRoutes(as, events, se).service, "/api/seqexec/commands")
+      .mountService(new SeqexecUIApiRoutes(as, events, se).service, "/api")
       .start
   }
 
   /**
     * Reads the configuration and launches the web server
     */
-  override def server(args: List[String]): Task[Server] =
-    for {
-      _    <- configLog
-      ac   <- authConf
-      wc   <- serverConf
+  override def server(args: List[String]): Task[Server] = {
+    val engineTask = for {
       c    <- config
       seqc <- SeqexecEngine.seqexecConfiguration.run(c)
-      as   <- authService.run(ac)
-      // Put the queue in WebServerConfiguration?
-      q    = async.boundedQueue[engine.Event](10)
-      ws   <- webServer(as, q, SeqexecEngine(seqc)).run(wc)
-    } yield ws
+    } yield SeqexecEngine(seqc)
+
+    val inq  = async.boundedQueue[engine.Event](10)
+    val out  = async.topic[SeqexecEvent]()
+
+    engineTask flatMap (
+      se => Nondeterminism[Task].both(
```

Comment:
That maybe related to the `ExecutorService` being used. It maybe that you've exhausted the thread counts so it cannot `fork` another one.

We've discovered that in general is not a good idea to rely on the default one as it uses 2*N (N= processors count) and thus gets exhausted sometimes